### PR TITLE
fixed bug for base64url

### PR DESCRIPTION
**Pull request by beenotung**
fix bug caught by below test:
```typescript
import {decode, encode, name} from 'typestub-multibase';
import {pki} from 'node-forge';

function randomString(length: number): string {
  let s = '';
  for (; s.length < length;) {
    s += Math.random()
      .toString(36)
      .split('.')[1];
  }
  return s.substring(0, length);
}

function toBuffer(b: Buffer | Uint8Array): Buffer {
  if (b instanceof Uint8Array) {
    return Buffer.from(b.buffer);
  }
  return b;
}

function test(name: name, pass: number) {
  for (let i = 0; i < pass; i++) {
    let keypair = pki.ed25519.generateKeyPair({seed: randomString(32)});
    let b = toBuffer(keypair.publicKey);
    try {
      decode(encode(name, b))
    } catch (e) {
      console.error({b});
      throw e;
    }
    let x = decode(encode(name, toBuffer(keypair.publicKey)).toString()).toString();
    let y = decode(encode(name, toBuffer(keypair.privateKey)).toString()).toString();
  }
  console.log(`passed ${pass} check on ${name}`);
}

try {
  test('base64', 1000);
} catch (e) {
  console.error('Failed test on base64', e);
}
try {
  test('base64url', 1000);
}
catch (e) {
  console.error('Failed test on base64url', e);
}
```

### CODE REVIEW

File: src/base64.js [hunk 1]
```diff
@@ -33,12 +33,13 @@ module.exports = function base64 (alphabet) {
...
       if (url) {
-        input = input.replace('+', '-')
-        input = input.replace('/', '_')
+        input = input.replace(/\+/g, '-')
+        input = input.replace(/\//g, '_')
       }
 
       for (let char of input) {
         if (alphabet.indexOf(char) < 0) {
+          console.error({char, input});
```
Comment (naure) [inline src/base64.js]:
```suggestion
```

**Comment by naure** [comment]
Can someone please merge this?

The CI error is just a full disk, nothing to do with the code.

**Comment by alanshaw** [comment]
@beenotung would you mind adding a test for this to the test suite?

**Comment by olizilla** [comment]
Thanks for catching this @beenotung. It's been merged in with tests as part of #39

**Pull request closed without merging**